app/decorators/service_provider_session.rb

```diff
@@ -132,9 +122,9 @@ def current_user
     view_context&.current_user
   end
 
-  private
-
-  attr_reader :sp, :view_context, :sp_session, :service_provider_request
+  def sp_aal
+    sp&.default_aal || 1
+  end
 
   def resolved_authn_context_result
```

Comment:
We should be able to remove this from the decorator, correct?

Reply:
Nope; it's used in `ServiceProviderSession#remember_device_default`. But I did make it private, to get it out of the public API.

Reply:
I am referring to the `#resolved_authn_context_result` method here. It looks like `#remember_device_default` is only using `#sp_aal` which is using `#sp` which you moved out of the public API.

https://github.com/18F/identity-idp/blob/947bbbfe2e9bd0a3e488a2a40a4d6d0120a219c6/app/decorators/service_provider_session.rb#L16-L18

Reply:
Oh. Yeah. Removed.